tpr/kind.go

```diff
@@ -0,0 +1,40 @@
+package tpr
+
+import (
+	"strings"
+
+	microerror "github.com/giantswarm/microkit/error"
+)
+
+/*
+	Code based on:
+	https://github.com/kubernetes/kubernetes/blob/6a4d5cd7cc58e28c20ca133dab7b0e9e56192fe3/pkg/registry/extensions/thirdpartyresourcedata/util.go
+*/
+
+// extractKindAndGroup extracts kind and group from a name. For details see
+// Config.Name godoc.
+func extractKindAndGroup(name string) (kind string, group string, err error) {
+	parts := strings.Split(name, ".")
+	if len(parts) < 3 {
```

Comment:
Can we leave it like that? IDK how to name it, and it makes sense for me to leave it, because it's immediately explained in error annotation below.